doc/source/toolchain.rst

```diff
@@ -0,0 +1,226 @@
+===================   ==============   ===================
+
+
+
+C and C++ Language Standards
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+C and C++ language standards for SciPy are generally guidelines
+rather than official decisions. This is particularly true of
+attempting to predict adoption timelines for newer standards.
+
+================  ===========================================
+ Date              C Standard
+================  ===========================================
+ <= 2018           C90
+ 2019              C90 for old code, may consider C99 for new
+ 2020              C99
+ ?                 C11
+ ?                 C17, C18
+================  ===========================================
+
+C99 has been a standard for 20 years, but experience has shown that
+not all features are supported equally well across all platforms.
+C18 is a bug fix for C11, so C11 may be skipped entirely.
+
+For C++, we currently suggest that anything beyond C++11 is going to be impossible
+for a very long time because of ecosystem support restrictions. See [4]_.
+
+.. note::
+
+    Developer Note: Some C99 features would be useful for scientific programming, in particular better support of
+    IEEE 754 [5]_.  SciPy has a small include file ``scipy/_lib/_c99compat.h`` which provides
+    access to a few functions.  Use in conjunction with ``<numpy/npy_math.h>``.
+
+    =========================================  ========================================================
+     Feature                                   Workaround
+    =========================================  ========================================================
+    ``isnan()``, ``isinf()``, ``isfinite()``   Use ``sc_isnan()``, ``sc_isinf()``, ``sc_isfinite()``
+    ``NAN``                                    Use ``NPY_NAN`` (*almost* equivalent)
+    inline functions                           Make static functions and place in an include .h file
+    mid-block variable declarations            Declare variables at the top of the block
+    =========================================  ========================================================
+
+
+Fortran Compilers
+^^^^^^^^^^^^^^^^^
+
+Generally, any well-maintained compiler is likely suitable and can be used to build SciPy.
+
+========  ==================
+ Tool      Version
+========  ==================
+gfortran    >= 4.8.0
+ifort      A recent version
+flang      A recent version
+========  ==================
+
+
+Cython Compiler
+^^^^^^^^^^^^^^^
+
+SciPy always requires a recent Cython compiler.
+
+========  ============  ===============
+ Tool     Tool Version   SciPy version                                                z
+========  ============  ===============
+Cython      >= 0.29.1    1.12.1
+========  ============  ===============
+
+
+
+Other Libraries
+^^^^^^^^^^^^^^^
+
+========  ===================
+ Tool      Minimum version
+========  ===================
+LAPACK      3.4.1.
+OpenBLAS    A recent version
```

Comment:
OpenBLAS can be removed in the table I think. Perhaps a sentence below about OpenBLAS, MKL, ATLAS being okay, and Accelerate no longer being supported.

Reply:
Done.